src/Areas/Marketplace/Services/MarketplaceService.cs

```diff
@@ -0,0 +1,174 @@
+
+        string queryString = string.Join("&", queryParams);
+        return $"{ManagementApiBaseUrl}/subscriptions/{subscription}/providers/Microsoft.Marketplace/products/{productId}?{queryString}";
+    }
+
+    private async Task<string> GetMarketplaceResponseAsync(string url, string? partnerTenantId,
+        string? pricingAudience, string? tenant, RetryPolicyOptions? retryPolicy = null)
+    {
+        // Use Azure Core pipeline approach consistently
+        var clientOptions = AddDefaultPolicies(new MarketplaceClientOptions());
+
+        // Configure retry policy if provided
+        if (retryPolicy != null)
+        {
+            clientOptions.Retry.MaxRetries = retryPolicy.MaxRetries;
+            clientOptions.Retry.Mode = retryPolicy.Mode;
+            clientOptions.Retry.Delay = TimeSpan.FromSeconds(retryPolicy.DelaySeconds);
+            clientOptions.Retry.MaxDelay = TimeSpan.FromSeconds(retryPolicy.MaxDelaySeconds);
+            clientOptions.Retry.NetworkTimeout = TimeSpan.FromSeconds(retryPolicy.NetworkTimeoutSeconds);
+        }
+
+        // Create pipeline
+        var pipeline = HttpPipelineBuilder.Build(clientOptions);
+
+        string accessToken = await GetAccessTokenAsync(tenant);
+
+        var request = pipeline.CreateRequest();
+        request.Method = RequestMethod.Get;
+        request.Uri.Reset(new Uri(url));
+        request.Headers.Add("Authorization", $"Bearer {accessToken}");
+
+        // Add optional headers as specified in the API
+        if (!string.IsNullOrEmpty(partnerTenantId))
+            request.Headers.Add("PartnerTenantId", partnerTenantId);
+
+        if (!string.IsNullOrEmpty(pricingAudience))
+            request.Headers.Add("PricingAudience", pricingAudience);
+
+        var response = await pipeline.SendRequestAsync(request, CancellationToken.None);
+
+        if (!response.IsError)
+        {
+            return response.Content.ToString();
+        }
+
+        throw new HttpRequestException($"Request failed with status {response.Status}: {response.ReasonPhrase}");
+    }
+
+    private async Task<string> GetAccessTokenAsync(string? tenant = null)
+    {
+        if (_cachedAccessToken != null && DateTimeOffset.UtcNow < _tokenExpiryTime)
+        {
+            return _cachedAccessToken;
+        }
+
+        AccessToken accessToken = await GetEntraIdAccessTokenAsync(ManagementApiBaseUrl, tenant);
+        _cachedAccessToken = accessToken.Token;
+        _tokenExpiryTime = accessToken.ExpiresOn.AddSeconds(-TokenExpirationBuffer);
+
+        return _cachedAccessToken;
+    }
+
+    private async Task<AccessToken> GetEntraIdAccessTokenAsync(string resource, string? tenant = null)
+    {
+        var tokenRequestContext = new TokenRequestContext([$"{resource}/.default"]);
+        var tokenCredential = await GetCredential(tenant);
+        return await tokenCredential
+            .GetTokenAsync(tokenRequestContext, CancellationToken.None)
+            .ConfigureAwait(false);
```

Comment:
ConfigureAwait is not necessary, there is no synchronization context to continue on.